### PR TITLE
Add support for amazon linux

### PR DESCRIPTION
**Pull request by pkallos**
also add broadcast_rpc_address params

### CODE REVIEW

**Comment by pkallos** [comment]
hmm what happened to the comments on this pull request? did I dream them?

**Comment by msimonin** [comment]
Hi @pkallos,
They are here https://github.com/pkallos/puppet-cassandra/commit/ccdfb02d730c284f177c8b6c3fa1c134184ea678#commitcomment-9329063 

Did you amend your commit or rebase at some point ? The commit id changed.

**Comment by pkallos** [comment]
Ah right, I did amend:

>We should maybe move the logic to generate the right package name somewhere else. 
> What do you think ?
>This way this could be more consistent to what we have for the cassandra package name.

After looking at this a bit more, the "cassandra package name" is configured alongside a bunch of other variables that are very specific to cassandra (they are all prefixed with cassandra_ in config.pp) so I'm wondering if moving "python_package_name" and "java_package_name" into there is the right move. Not clear.

**Comment by msimonin** [comment]
What about putting those in specific files like ```config/java.pp``` and ```config/python-cql.pp``` ?

**Comment by pkallos** [comment]
> What about putting those in specific files like config/java.pp and config/python-cql.pp ?

Ok, I dropped the specifics for installing java and python-cql into `install/java.pp` and `install/python-cql.pp`, because it seemed to me like `config/` was responsible for user groups, configuration files etc... Whereas this is specifically about install packages.

**Comment by pkallos** [comment]
Found an other issue when testing with `tests/Vagrantfile` on ubuntu, there was an issue where the apt-key wasn't being added. Also confirmed #9 happening for me.

File: manifests/install.pp [hunk 1]
```diff
@@ -1,11 +1,8 @@
 class cassandra::install {
 
-    if !defined (Package['java']) {
-      package { 'java':
-        ensure  => installed,
-        name    => 'openjdk-7-jre'
-      }
-    }
+    class { 'cassandra::install::java': }
```
Comment (msimonin) [inline manifests/install.pp]: why not using ``ìnclude```?

Comment (pkallos) [inline manifests/install.pp]: update

**Comment by pkallos** [comment]
:+1:

**Comment by msimonin** [comment]
@pkallos, thank you for your contribution !
Actually I've something to propose.
I'm thinking to rewrite the module from scratch, they are a bunch of legacy code wich will be hard to maintain in the future (issue #10 is a good example )
From my side I think this module should only support the last stable version of cassandra on a various number of platforms (red hat, debian, amazon ... ) like this module does : https://github.com/puppetlabs/puppetlabs-rabbitmq

If you want we can bootstrap a new project on a dedicated branch.

**Comment by pkallos** [comment]
Sounds good, though I am not sure I will be able to contribute much, I am very new to puppet at the moment and I'm mostly using it as a means to an end to focus my work on Cassandra.

**Comment by pkallos** [comment]
anything else on this? would be nice to get this in the next puppet module release! Thanks

**Comment by pkallos** [comment]
@msimonin :+1:/:-1: ?

**Comment by msimonin** [comment]
Definitely :+1: my only concern is about what was fixed in #9 . We probably need to anchor as well each of the class `cassandra::install::java` and `cassandra::install::python-cql`.

**Comment by Poil** [comment]
Java on EL6 also need the same case statement :+1: 

``` ruby
    if !defined (Package['java']) {
      case $::osfamily {
        'Debian': { $jre_package = 'openjdk-7-jre' }
        'RedHat': { $jre_package = 'java-1.7.0-openjdk' }
        default: { fail("Error : ${module_name} unsuported OSFamily") }
      }
      package { 'java':
        ensure  => installed,
        name    => $jre_package
      }
    }
```

**Comment by pkallos** [comment]
closing from staleness, feel free to reopen

**Pull request closed without merging**